Make video info section more concise

jasonhenriquez opened the pull request:
# Make video info section more concise

<!-- Thanks for sending a pull request! Make sure to follow the contributing guidelines. -->
<!-- Important note, we may remove your pull request if you do not use this provided PR template correctly. -->

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [x] Bugfix
- [x] Feature Implementation

## Related issue
closes #4336 

## Description
- Puts title on its own row
- Puts video info section on its own row
- Puts subscribe button & video buttons on their own row
- Fixes bug of `videoOptions` buttons overflowing the video info section that occurred on some viewport widths

## Video <!-- If appropriate -->

https://github.com/FreeTubeApp/FreeTube/assets/84899178/18ce83a6-c7e9-4001-9fe5-d6e6ea4a3711



## Testing <!-- for code that is not small enough to be easily understandable -->
- Ensure that appearance is great at all screen sizes
- Ensure that the video options buttons no longer extend past the container when resizing viewport

## Desktop
<!-- Please complete the following information-->
- **OS:** OpenSUSE Tumbleweed
- **OS Version:** 2023xxxx
- **FreeTube version:** 0.19.1

## Additional context
<!-- Add any other context about the pull request here. -->

absidue commented on src/renderer/components/watch-video-info/watch-video-info.scss:
```diff
@@ -47,38 +49,30 @@
-  }
-}
+.videoMetrics {
+  font-size: 12px;
```
Please choose a larger font size, 16px is considered the minimum that is readable, especially as the items, that are now on that row, used to be 16px and 15px before (stylelint used to complain about that but afaik you disabled everything, including that check, when you did you block and inline size rules).

jasonhenriquez replied:
> 16px is considered the minimum that is readable

Is it okay if I ask what your source is for this claim? The closest I can find is [this article](https://fontsaga.com/why-does-font-size-matter-in-design/) recommending 16 pixels as optimal for mobile, which ends up being fine here as well, as device manufacturers use the reference px size to set the actual rasterizing size based on intended distance of viewing. Not to uphold YouTube as a superior example of things, but they themselves use 12px font for the "1.46M subscribers" text in the image below. For the # of views and likes, they use 14px font. From my research, [12px seems to be the consistent minimum recommended font size](https://accessibility.psu.edu/fontsizehtml/) (16px should generally be used as the minimum for body text, and 12px is acceptable for supplementary information). If we had a stylelint rule that disallowed font sizes below 16px (which seemingly wasn't being enforced, as I see dozens of instances of 12px and even 11px font-sizes in our codebase), I personally think that's unnecessarily stringent. It's useful in places like here for demarcating levels of importance of information and guiding the eye to salient focal points.

![Screenshot_20231115_153641](https://github.com/FreeTubeApp/FreeTube/assets/84899178/1e0a8e1d-36ab-4e3e-9220-e46a0f729451)

absidue replied:
The stylint checks were optional for so long because we had many different violations in the codebase. Anyway here is the markdown document associated with that accessibility rule:

https://github.com/double-great/stylelint-a11y/tree/main/src/rules/font-size-is-readable

jasonhenriquez replied:
The articles cited there in the justification for the rule are the MDN article for `font-size` (which doesn't seem to broach the topic of recommended minimum size) and [this article talking about body text](https://marvelapp.com/blog/body-text-small/). When we add this rule back, we should definitely consider adding it as a warning, as there are quite a few exceptions to that rule.

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (dismissed):
LGTM

PikachuEXE commented:
I find it weird to see like number+icon left alone on the right now
Feels like it can be moved to video info row

jasonhenriquez commented:
@PikachuEXE Thank you, will consider it if others prefer as well - that is YouTube's approach that was considered for the main issue, but I don't really have a super strong preference

PikachuEXE commented:
It's not blocking especially if it's following YT
Just feel like it look strange now especially with wide window

PikachuEXE commented:
Maybe the breakpoint can be set at a greater width?
I see this with 901px width
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/0d62530d-4642-4d7f-93a8-9ab37dd7bc06)

jasonhenriquez commented:
@PikachuEXE The breakpoint at which "Up Next" is put on its own level?

PikachuEXE commented:
Sorry for being unclear, it's the buttons now arranged (squeezed) into a square

jasonhenriquez commented:
Are you talking about setting it to wrap into a box earlier? If you mean the other way around, that is difficult to do without using shrinking the button size, as this is just the `flex` behavior. What happens when it doesn't compress is this existing bug mentioned in the 4th bullet point in the description (taken from nightly):

![Screenshot_20240130_171950](https://github.com/FreeTubeApp/FreeTube/assets/84899178/dbc7baf1-5a68-4ebb-8d7e-d1bc0cbeb3c7)

PikachuEXE commented:
> Are you talking about setting it to wrap into a box earlier?

As I said `Maybe the breakpoint can be set at a greater width?` (e.g. 1000px)

jasonhenriquez commented:
Sorry, I'm having trouble parsing what you mean. It's `flex`, so it's just sizing dynamically based off of the space available.

PikachuEXE commented:
I mean the breakpoints for different panels in watch page (`src/renderer/views/Watch/Watch.scss`)
(Probably replace all `min-width: 901px` w/ `min-width: 1001px`

1001px
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/e7c22787-f223-4f70-9fac-6ae2bfc69810)
999px
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/86f493c9-0675-43cf-9ec4-25facc1085a9)

Also I think this should be rebased/merged with dev to be able to test probably due to new buttons

jasonhenriquez commented:
Ah, I am terrible at reading, it seems! Increased it to 1000px from 900px.

> Also I think this should be rebased/merged with dev to be able to test probably due to new buttons

Did this as well now

PikachuEXE commented:
A few more pixels would be great :)
at 1005px:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/67197a92-616d-45a3-8271-c1a6fd9d654d)

jasonhenriquez commented:
Blerg, good callout. I just made it 1050px to compensate for other languages with a longer word for "Unsubscribe" or "Subscribe"

PikachuEXE commented:
community-post updated too?

jasonhenriquez commented:
I modified the wrong `likeCount` from a different component & didn't realize it because I added the change in my local template. Not my brightest day...

PikachuEXE reviewed (dismissed):
Look good at 105x px

PikachuEXE commented:
Oh wait, I enabled theatre mode and the breakpoint is not updated for it...
977px:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/9859f7e8-64ab-484f-9ba7-629fda7b255b)

jasonhenriquez commented:
And it's even narrower if the sidebar is fully expanded. I don't think we can fully prevent the button container from wrapping early under these scenarios without picking an excessively large breakpoint

PikachuEXE commented:
I am saying that the breaking point for theatre mode should **also** be `1050px` (since breakpoint for non theatre mode updated)
(The one with `&.useTheatreMode {` & `@include theatre-mode-template;`)
1051px:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/ab51ef68-49de-490c-bc6d-55ab48fa467c)
1047px:
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/2411e28d-8970-4134-a714-1dc90cc9c24f)
